manifests/config.pp

```diff
@@ -65,4 +65,24 @@
       path        => ['/usr/local/bin', '/usr/local/sbin', '/usr/bin', '/usr/sbin', '/bin', '/sbin'],
     }
   }
+
+  if $pam_pkcs11::pam_config == 'pam-auth-update' {
+    file { '/usr/share/pam-configs':
+      ensure => directory,
+      mode   => '0755',
+    }
+
+    file { 'pkcs11':
+      ensure  => file,
+      path    => '/usr/share/pam-configs/pkcs11',
```

Comment:
The only reason was that it matches the pattern of the other `file` resources in this file. I wanted to stay consistent with what is done before. But I guess we can change that. I don't really have an opinion about this way or another here.

Reply:
Ah, I see. I don't see the advantage of having a separate name for the resource if that is not used; in the best case, it's just unnecessary, in the worst it might [cause conflicts](https://puppet.com/docs/puppet/7/lang_resources.html#lang_resources-resource-uniqueness).

Reply:
Well I guess it was just for convenience sake. So that there is no need to specify the full path whenever that resource is addressed.

I created a patch which converts the name of all file resources to use the full absolute path instead.